### ISSUE TITLE
Default `withMiddlewareAuthRequired()` usage not considering custom NextJS basePath

### ISSUE THREAD

**Issue by l-campbell** [auth0/nextjs-auth0#1118]
### Checklist

- [X] The issue can be reproduced in the [sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [README](https://github.com/auth0/nextjs-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

## Setting the scene

A new NextJS application with a custom `basePath` NextJS config value i.e. `/foo`. 
Also adding a `NEXT_PUBLIC_AUTH0_LOGIN` environment variable with the value of `/foo/api/auth/login`.

## What's happening?

I added the default scaffold `withMiddlewareAuthRequired()` in middleware:

```bash
export default withMiddlewareAuthRequired();
```

Running the application and navigating to `http://localhost:3000/foo/api/auth/login` the middleware returns a 401 JSON response to the browser window, instead of redirecting to universal login.

```bash
{
  "error": "not_authenticated",
  "description": "The user does not have an active session or is not authenticated"
}
```

The `pathname` in the `withMiddlewareAuthRequiredFactory()` function does not contain the `basePath` value so the function believes the incoming request is an API request, `pathname` equaling `/api/auth/login`. See source [here](https://github.com/auth0/nextjs-auth0/blob/12e979c886c64a17ebf2a802c1370fba217425f8/src/helpers/with-middleware-auth-required.ts#L69).

The custom login URL is being used the `ignorePaths` check earlier in the function, thus not returning early.

## Expected to happen

The response does not return a JSON payload and redirects to universal login.

### Reproduction

1. Create new NextJS app with `@auth0/nextjs-auth0` integrated
2. Add `basePath: '/foo'` value to `next.config.js`
3. Add `NEXT_PUBLIC_AUTH0_LOGIN='/foo/api/auth/login'` environment variable to `.env.local`
4. Run the application
5. Navigate to `http://localhost:3000/foo/api/auth/login`

### SDK version

v2.3.1

### Next.js version

v13.2.4

### Node.js version

v18.15.0

**Comment by adamjmcgrath** [comment]
Hi @l-campbell - thanks for raising this

You should not need to update `NEXT_PUBLIC_AUTH0_LOGIN` to accommodate the Next base path - there's a couple of places you'll need to update, but not this env var, see https://github.com/auth0/nextjs-auth0#base-path-and-internationalized-routing for full details

Do you run into any issues when you don't set `NEXT_PUBLIC_AUTH0_LOGIN`?

**Comment by adamjmcgrath** [comment]
Hi @l-campbell 

Could you use the edge instance and configure the login url this way:

```js
import { initAuth0 } from '@auth0/nextjs-auth0/edge';

const auth0 = initAuth0({
  routes: {
    login: '/foo/api/auth/login',
    callback: '/foo/api/auth/callback',
    postLoginRedirect: '/foo
  }
});

export default auth0.withMiddlewareAuthRequired();
```

**Comment by zcarlson-gmgt** [comment]
Hi @adamjmcgrath! Thanks for that edge instance suggestion, I think this will prove useful to us. Is it still best practice to export the edge instance and import in other areas such as the route handler? Just double-checking the edge instance implementation with middleware vs the server/node instance implementation.

**Comment by adamjmcgrath** [comment]
Hi @zcarlson-gmgt - np, you need a separate instance per runtime - so if your route handler uses the node runtime you should use `import { initAuth0 } from '@auth0/nextjs-auth0';`

**Comment by eyouth** [comment]
Hi guys, I got this particular problem as well, the issue is that I added a vercel.json file. since I am using vercel CLI, I added it to the vercelignore file and deploy a new one. Eventually that solved my problem. 

Hope that helps.

**Comment by Dalimil** [comment]
@l-campbell you are correct that it is currently broken. Here's a temporary workaround until this is fixed:

```ts
const edgeAuth0 = initAuth0(auth0Config);
const login = '/api/auth/login';
const callback = '/api/auth/callback';

// We cannot use the default edgeAuth0.withMiddlewareAuthRequired
// because of this issue in the library: https://github.com/auth0/nextjs-auth0/issues/1118
// The issue comes from the fact that req.pathname doesn't contain base url prefix so the URL redirect is invalid.
export async function middleware(req: NextRequest) {
  const { pathname, origin, search } = req.nextUrl;

  // Skip auth check for certain paths
  const ignorePaths = [login, callback, '/_next', '/favicon.ico', '/manifest.json'];
  if (ignorePaths.some((p) => pathname.startsWith(p))) {
    return;
  }

  const res = NextResponse.next();
  const session = await edgeAuth0.getSession(req, res);
  // Handle unauthenticated user
  if (!session?.user) {
    if (pathname.startsWith('/api')) {
      return NextResponse.json(
        {
          error: 'not_authenticated',
          description: 'The user does not have an active session or is not authenticated',
        },
        { status: 401 }
      );
    }
    const basePath = auth0Config.baseURL?.replace(origin, '');
    const returnTo = `${basePath}${pathname}${search}`;
    return NextResponse.redirect(new URL(`${basePath}${login}?returnTo=${encodeURIComponent(returnTo)}`, origin));
  }
  return res;
}
```

**Comment by matthew-gladman-oua** [comment]
> @l-campbell you are correct that it is currently broken. Here's a temporary workaround until this is fixed:

We had to do a very similar workaround as well, can confirm it is a bug

**Comment by mcgoooo-temporary** [comment]
i would also check whethere in your case (like mine) this is caused by next.config and trailing slash

**Comment by l-campbell** [comment]
@adamjmcgrath (sorry for the tag on a closed issue) but I do believe this is still an issue, do you mind taking another look at?

**Comment by mjqlawson** [comment]
Just spent two days trying to figure this out with "@auth0/nextjs-auth0": "^3.5.0", the workaround posted Dalimi is the only thing that worked for us. @adamjmcgrath, I'm with @l-campbell could you take another look?
